sdk/core/core-http/lib/restError.ts

```diff
@@ -14,13 +14,25 @@ export class RestError extends Error {
   request?: WebResource;
   response?: HttpOperationResponse;
   body?: any;
-  constructor(message: string, code?: string, statusCode?: number, request?: WebResource, response?: HttpOperationResponse, body?: any) {
+  parsedBody?: unknown;
+  parsedHeaders?: { [key: string]: unknown };
+  
+  constructor(
+    message: string,
+    code?: string,
+    statusCode?: number,
+    request?: WebResource,
+    response?: HttpOperationResponse,
+    body?: any,
+    parsedHeaders?: { [key: string]: unknown }
```

Comment:
I would skip making the change to the constructor as part of this PR.
If in the future we feel the need to set this when constructing the object, we can always add this later.
Right now, we dont have the requirement to do so.

Reply:
Done